cmd/dep/init.go

```diff
@@ -152,7 +163,20 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	l = dep.LockFromSolution(soln)
 
 	rootAnalyzer.FinalizeRootManifestAndLock(m, l)
-	gs.FinalizeRootManifestAndLock(m, l)
+	if cmd.gopath {
+		gs.FinalizeRootManifestAndLock(m, l)
+	} else {
+		// Pick the direct dependencies from the above solution and add to manifest
```

Comment:
IIRC, this was done to generate the final lock memo with the obtained solution.
When there's no external tool and -gopath is not passed, we use an empty manifest in the solver to get a solution. Then we use the lock obtained from the solution and pick direct dependencies from there and create a new manifest, using which we generate the final lock memo.

Now, in case of an external tool, our initial manifest may or may not be empty. So we use the solution lock to fill up the manifest with direct dependencies. In case, manifest was empty, it gets filled. And in case, manifest already had packages, there wouldn't be any change. Or if there were any missing direct dependencies in manifest, they get added too.

Hope this answers your question.